File: 1-5/solution.py
#!/usr/bin/python
def isReplace(s1,s2):
	replace=0	
	for i in range(len(s1)):
		if s1[i]==s2[i]:
			continue
		if replace==0:
			replace=1
		else:
			return "NO"
	return "YES"

def isInsert(s1,s2):
	insert=0
	j=0
	for i in range(len(s1)):
		if s1[i]==s2[j]:
			j=j+1
		elif insert==0:
			insert=1
		else:
			return "NO"
	return "YES"




s1="pale"
s2="ple"


if len(s2)>len(s1):
	tmp=s2	
	s2=s1
	s1=tmp

if len(s1)-len(s2)>1:
	print "NO"
	quit()

if len(s1)==len(s2):
	print isReplace(s1,s2)
	quit()

print isInsert(s1,s2)

File: 1-5/solution.cpp
#include<iostream>
#include<string>


using namespace std;

string isReplace(string s1,string s2);
string isInsert(string s1, string s2);

int main()
{
	string s1 = "pale";
	string s2 = "bake";	
	
	if (s2.size()>s1.size())
	{	
		string tmp=s2;
		s2=s1;
		s1=tmp;
	}
	
	if(s1.size()-s2.size()>1)
	{
		cout << "FALSE" << endl;
		return 0;
	}

	if(s1.size()==s2.size())
	{
		cout << isReplace(s1,s2) <<endl;
		return 0;
	}
	else
	{
		cout << isInsert(s1,s2) << endl;
		return 0;
	}
		
	
	return 0;
}



string isReplace(string s1,string s2)
{
	bool replace=0;
	for(int i=0;i<s1.size();i++){
		if(s1[i]==s2[i])
			continue;
		if (replace==0)
			replace=1;
		else
			return "NO";
	}	
	return "YES";
}


string isInsert(string s1, string s2)
{
	bool insert=0;
	int j=0;
	for(int i=0;i<s1.size();i++){
		if (s1[i]==s2[j])
			j++;
		else if (insert==0)
				insert=1;
			else
				return "NO";
	}	
	return "YES"	;
	
}


